x-pack/platform/plugins/shared/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -483,12 +483,29 @@ export const LensTopNavMenu = ({
     isOnTextBasedMode,
   ]);
 
+  const [hasShareIntegration, setHasShareIntegration] = useState(false);
+
   useEffect(() => {
+    let canceled = false;
```

Comment:
Maybe it would be better to have a reusable custom hook that different plugins could use to check if their integrations are available?